### PR TITLE
Extensions-api: initial hello-world example

### PR DESCRIPTION
**Pull request by ixrock**
Adds new icon and page in cluster-view

image

### CODE REVIEW

File: src/extensions/example-extension/example-extension.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+export default class ExampleExtension extends LensExtension {
+  protected unRegisterPage = Function();
+
+  onActivate() {
+    extension = this
+    console.log('EXAMPLE EXTENSION: ACTIVATE', this.getMeta())
+    const { dynamicPages } = this.runtime;
+
+    this.unRegisterPage = dynamicPages.register({
```
Comment (jakolehm) [inline src/extensions/example-extension/example-extension.tsx]: Maybe the API should be something like:

```js
this.registerPage({
      type: DynamicPageType.CLUSTER,
      path: "/extension-example",
// ...
})
```
and LensExtension would automatically do unregister etc housekeeping.

File: src/extensions/example-extension/example-extension.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { Button, DynamicPageType, Icon, LensExtension } from "@lens/extensions"; // fixme: map to generated types from "extension-api.ts"
+import React from "react";
+import path from "path";
+
+let extension: ExampleExtension; // todo: provide instance from context
```
Comment (Nokel81) [inline src/extensions/example-extension/example-extension.tsx]: Couldn't this be achieved by a static method on `ExampleExtension`?

Comment (ixrock) [reply src/extensions/example-extension/example-extension.tsx]: This solution was temporary

File: src/extensions/example-extension/example-extension.tsx (hunk 1 above)
Comment (ixrock) [reply src/extensions/example-extension/example-extension.tsx]: Done

**Review by jakolehm — APPROVED** [review]
I think this is a good start 👍 Let's improve API in follow-up PRs.

**Comment by jakolehm** [comment]
Tests are still failing...

**Pull request merged**